### ISSUE TITLE
[Question] Custom way to send data to server

### ISSUE THREAD

**Issue by saransh-malik** [transistorsoft/react-native-background-geolocation#2205]
## Your Environment
* Plugin version: 4.15.2
* Platform: iOS
* OS version: 17.2
* Device manufacturer / model: iPhone 15
* React Native version (`react-native -v`): 0.69
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->
Background.setConfig({
...other basic props
  url: `http://localhost:3000/locations`,
      headers: {
        Origin: 'https://mapmycustomers.me',
        'Content-Type': 'application/json',
        'Accept-Encoding': 'gzip',
        'x-compression': true,
        'x-mmc-client': 'Mobile',
        Authorization: `Bearer ${this.props.auth?.data?.accessToken}`,
      },
      params: { tripId: data?.id, id: data?.id },
      maxBatchSize: 10,
      httpRootProperty: 'data',
      autoSync: true,
      batchSync: true,
})
```

## Expected Behavior
I am looking to figure out a way to completely customise the request body format being sent to the server. As of now by default the library adds each row per location as different object inside an array and sends data to the server. What I am looking to achieve is a format like below: 
 
`{points: [[lat,long],[lat,long],[lat,long]], ...params}`

So basically instead of different object for each location, I want all locations in a nested array. 
I already tried to play around with the config but was not able to achieve it. 
Is something of this possible or I just have to rely on sending locations manually?

**Comment by christocracy** [comment]
See api docs Config.locationTemplate.

you’re not going to send the timestamp?

**Comment by saransh-malik** [comment]
@christocracy I did use that, but while it gets into my template it sends multiple arrays instead of a single one.
So this is what I get
`[ 
   [Array]],
    [Array]
],`

when I specify this in locationTemplate: 
`[[<%=latitude%>, <%=longitude%>],[<%=latitude%>, <%=longitude%>]]`

**Comment by saransh-malik** [comment]
@christocracy Actually sorry, I think I figured it out. Closing this for now.

**Comment by christocracy** [comment]
Crazy how you’re just ignoring the timestamp of when the location was recorded.  That doesn’t matter at all to you?

**Comment by saransh-malik** [comment]
> Crazy how you’re just ignoring the timestamp of when the location was recorded.  That doesn’t matter at all to you?
> 

Nope, location makes more relevance for us not the time it was recorded. Unless the library can mess up the order? 
I am still in initial stages of development so not sure if the order is always trustable or not. 
Or do you suggest it is always better to consider using the time as well.

**Comment by christocracy** [comment]
The plug-in sends data in the order received.

hiccups can always happen.  The location api can return a stale location (the location also has an `age` attribute, telling you how old it is.

you’ll find out the hard way.
